torch_xla/csrc/ops/ops_xla_shape_fn.cpp

```diff
@@ -421,6 +421,21 @@ xla::Shape GeTensorOutputShape(const torch::lazy::Value& self,
   return GeScalarOutputShape(self, other);
 }
 
+xla::Shape GluOutputShape(const torch::lazy::Value& input, int64_t dim) {
+  const xla::Shape& input_shape = GetXlaShape(input);
+
+  if (dim < 0) dim += input_shape.rank();
+
+  absl::Span<const int64_t> inp_dimensions = input_shape.dimensions();
+  std::vector<int64_t> output_sizes(std::begin(inp_dimensions),
+                                    std::end(inp_dimensions));
+ 
+  // Output shape is always half the input shape on the specified dimension
+  output_sizes[dim] = inp_dimensions[dim] / 2;
+ 
+  return xla::ShapeUtil::MakeShape(input_shape.element_type(), output_sizes);
+}
+
 xla::Shape GtScalarOutputShape(const torch::lazy::Value& self,
```

Comment:
can you add a cpp test similar to https://github.com/pytorch/xla/blob/028df4da388468fa9a41b1f98ea08bfce13b4c63/test/cpp/test_aten_xla_tensor_5.cpp#L209 ?

Reply:
There should be a cpp test already added or another one?

Reply:
you should add another one similar to Gelu.

Reply:
Hey Jack can you help me understand what is missing?  

The test in https://github.com/pytorch/xla/pull/5466/files#diff-0c11cc2fbeb0cfbc8fe223f9ac3df0904dcbb1277946a55a56cb102642f4faefR1322-R1342 was intended to be similar to the Gelu one, it tests a variety of input permutations across both input variables and uses the same checks.

Reply:
You can build it with `BUILD_CPP_TESTS=1` and run CPP test with `./test/cpp/run_test.sh`